File: clase-03-17/main.cpp
#include "tads/AVL.cpp"
#include <iostream>

int main(int argc, char const *argv[])
{
  ABBAbs<int> *abb = new AVL<int>();
  std::printf("Altura del arbol: %d\n", abb->height());
  std::printf("#Elementos del arbol: %d\n", abb->size());

  abb->insert(42);
  abb->insert(24);
  abb->insert(12);
  std::printf("Altura del arbol: %d\n", abb->height());
  std::printf("#Elementos del arbol: %d\n", abb->size());

  std::printf("Elementos:\n");
  abb->inOrderPrint();

  return 0;
}

File: clase-03-17/Main.java
import tads.ABBI;
import tads.AVL;
import java.lang.System;

public class Main {
  public static void main(String[] args) {
    ABBI<Integer> abb = new AVL<Integer>();
    System.out.printf("Altura del arbol: %d\n", abb.height());
    System.out.printf("#Elementos del arbol: %d\n", abb.size());

    abb.insert(42);
    abb.insert(24);
    abb.insert(12);
    System.out.printf("Altura del arbol: %d\n", abb.height());
    System.out.printf("#Elementos del arbol: %d\n", abb.size());

    System.out.printf("Elementos:\n");
    abb.inOrderPrint();

  }
}
